### ISSUE TITLE
Broken system registration state when the base product is not fully mirrored

### ISSUE THREAD

**Issue by ikapelyukhin** [SUSE/rmt#276]
First registration attempt:

```
# ./rmt-client-setup https://my-local-rmt.example.org
Error: SCC returned 'Not all mandatory repositories are mirrored for product SUSE Linux Enterprise Server 12 SP3 x86_64' (422)
```

Enabled the repos on RMT, second registration attempt:

```
# ./rmt-client-setup https://my-local-rmt.example.org
The system is already registered. Please de-register first by calling:
$> SUSEConnect --de-register
$> SUSEConnect --cleanup
```

At this point turns out that `/etc/zypp/credentials.d/SCCcredentials` file was written, however `/etc/SUSEConnect` with RMT server URL wasn't, so de-registration isn't possible:

```
# ./bin/SUSEConnect -d
Error: Invalid system credentials, probably because the registered system was deleted in SUSE Customer Center. Check https://scc.suse.com whether your system appears there. If it does not, please call SUSEConnect --cleanup and re-register this system.
```

We could check if the product is fully mirrored during `announce` API call (the base product data should be present there) and raise the error. Alternatively, this could be fixed by forcing SUSEConnect to write `/etc/SUSEConnect` even if an error occurs.

**Comment by felixsch** [comment]
@ikapelyukhin Is there a easy way to determine if a repo/product is fully mirrored?

**Comment by ikapelyukhin** [comment]
@felixsch well... This message is generated somehow: `'Not all mandatory repositories are mirrored for product SUSE Linux Enterprise Server 12 SP3 x86_64'` -- so I guess there is. :slightly_smiling_face: 

This probably should be handled on SUSEConnect side, it should clean up the configs/credentials on the client system if the transaction fails in the middle.

**Comment by tmuntaner** [comment]
This should be fixed by both SUSE/rmt#610 and SUSE/connect#440.
